core/emulate.c

```diff
@@ -292,6 +310,58 @@ static const struct em_opcode_t opcode_table_0F3A[256] = {
     X16(N), X16(N), X16(N), X16(N),
 };
 
+static uint64_t gpr_read_shifted(struct em_context_t *ctxt,
+                                 unsigned index, size_t size, size_t shift)
+{
+    uint64_t value = 0;
+    uint8_t *value_ptr;
+
```

Comment:
nit: Add `assert()` statements to detect potential buffer overflows:

* `index < 16`
* `size + shift <= 8`